### PR TITLE
Allow fio commands to accept VFS paths like vfs://datasource

### PR DESCRIPTION
**Pull request by geowurster**
Closes https://github.com/Toblerity/Fiona/issues/263

Ready for review.  Changes are mostly just removing `type=click.Path()` for all the input `@click.argument()`s.

### CODE REVIEW

File: fiona/fio/cat.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 # Cat command
 @click.command(short_help="Concatenate and print the features of datasets")
-@cligj.files_in_arg
```
Comment (geowurster) [inline fiona/fio/cat.py]: This uses `click.Path()` which chokes on the VFS paths, so I copied the source and removed the type declaration.

File: tests/test_fio_dump.py [hunk 2]
```diff
@@ -28,3 +31,13 @@ def test_dump_layer():
         result = runner.invoke(dump.dump, [TESTGPX, '--layer', layer])
         assert result.exit_code == 0
         assert '"FeatureCollection"' in result.output
+
+
+def test_dump_layer_vfs():
+    path = 'zip://tests/data/coutwildrnp.zip'
+    result = CliRunner().invoke(dump.dump, [path])
```
Comment (sgillies) [inline tests/test_fio_dump.py]: Can you make this `CliRunner().invoke(main_group, 'dump', ...)`? That will help with moving the Fiona env to the main_group ctx someday like we did in Rasterio.

**Comment by geowurster** [comment]
Just noticed the tests are failing.  Will address later this weekend.

**Comment by geowurster** [comment]
I have no idea why, but the failing tests seem to be `nose` related.  The offender is `tests/test_fio_cat.py::test_multi_layer_fail`, which was introduced in https://github.com/Toblerity/Fiona/pull/354 to let `$ fio cat` handle multiple layers for multiple input files.  Travis gave the PR a green light across the board.

The test checks to make sure that `$ fio cat --layer 200000:coutlildrnp tests/data` produces an error, because the datasource `tests/data` has no layer index `200000`.

When executing from the command line:

```console
$ fio cat --layer 200000:coutlildrnp tests/data
Usage: fio cat [OPTIONS] INPUTS...

Error: Invalid value: Layer 200000:coutlildrnp does not exist
```

When executing with `pytest`:

```console
$ py.test tests/test_fio_cat.py::test_multi_layer_fail
====================================================================== test session starts =======================================================================
platform darwin -- Python 3.5.1, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/kwurster/code/Fiona, inifile: 
plugins: cov-2.3.1
collected 9 items 

tests/test_fio_cat.py .

==================================================================== 1 passed in 0.08 seconds ====================================================================
```

But when executing with `nosetests` the test just sits and does nothing and then Travis times out after 10 minutes.  Interestingly the `nosetests` command issues a few warnings, the last of which is related to `inspect.getargspec()` and is issued in `nose/util.py`:

```console
$ nosetests tests.test_fio_cat:test_multi_layer_fail
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
/Users/kwurster/code/Fiona/tests/fixtures.py:5: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kwurster/code/Fiona/tests/data/collection.txt' mode='r' encoding='UTF-8'>
  return open(os.path.join(os.path.dirname(__file__), name)).read()
/Users/kwurster/code/Fiona/tests/fixtures.py:5: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kwurster/code/Fiona/tests/data/collection-pp.txt' mode='r' encoding='UTF-8'>
  return open(os.path.join(os.path.dirname(__file__), name)).read()
/Users/kwurster/code/Fiona/tests/fixtures.py:5: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kwurster/code/Fiona/tests/data/sequence.txt' mode='r' encoding='UTF-8'>
  return open(os.path.join(os.path.dirname(__file__), name)).read()
/Users/kwurster/code/Fiona/tests/fixtures.py:5: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kwurster/code/Fiona/tests/data/sequence-pp.txt' mode='r' encoding='UTF-8'>
  return open(os.path.join(os.path.dirname(__file__), name)).read()
/Users/kwurster/code/Fiona/venv/lib/python3.5/site-packages/nose/util.py:453: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() instead
  inspect.getargspec(func)
tests.test_fio_cat.test_multi_layer_fail ...
```

Python and Nose versions:

```console
$ python --version
Python 3.5.1
$ nosetests --version
nosetests version 1.3.7
```

Any ideas?  Can someone confirm?  The switch to `pytest` in https://github.com/Toblerity/Fiona/pull/376` is almost ready so maybe that's the right solution.

**Comment by sgillies** [comment]
@geowurster no idea. I don't feel like it's necessary to solve a nose-specific issue if switching to pytest works.

**Comment by geowurster** [comment]
Appveyor strikes again.  @sgillies could you add me to the Appveyor project?  I created an account associated with my GitHub account and my personal email address.

**Comment by sgillies** [comment]
@geowurster done, but I re-built and see the same error.

**Pull request merged**